bigchaindb_driver/driver.py

```diff
@@ -37,19 +37,41 @@ def __init__(self,
                  transport_class=Transport):
         """Initialize a :class:`~bigchaindb_driver.BigchainDB` driver instance.
 
+        If a :attr:`verifying_key` and :attr:`signing_key` are given, this
+        instance will be bound to the given keypair and be applied as defaults
+        whenever a verifying and/or signing key are needed.
+
         Args:
-            *nodes: BigchainDB nodes to connect to. Currently, the full URL
-                must be given In the absence of any node, the default of the
-                :attr:`transport_class` will be used, e.g.:
+            *nodes (str): BigchainDB nodes to connect to. Currently, the full
+                URL must be given. In the absence of any node, the default of
+                the :attr:`transport_class` will be used, e.g.:
                 ``'http://localhost:9984/api/v1'``.
-            verifying_key (str): the base58 encoded public key for the ED25519
-                curve.
-            signing_key (str): the base58 encoded private key for the ED25519
-                curve.
-            transport_class: Transport class to use. Defaults to
-                :class:`~bigchaindb_driver.transport.Transport`.
+            verifying_key (str, keyword, optional): the base58 encoded public
+                key for the ED25519 curve to bind this driver with.
+            signing_key (str, keyword, optional): the base58 encoded private
+                key for the ED25519 curve to bind this driver with.
+                Must be given with :attr:`verifying_key`.
+            transport_class (Transport, keyword, optional): Transport class to
+                use.
+                Defaults to :class:`~bigchaindb_driver.transport.Transport`.
 
+        Raises:
+            :class:`~bigchaindb_driver.exceptions.InvalidVerifyingKey: if the
+                given :attr:`verifying_key` was not valid, or a
+                :attr:`signing_key` was given without a :attr:`verifying_key`
+            :class:`~bigchaindb_driver.exceptions.InvalidSigningKey: if the
+                given :attr:`signing_key` was not valid
         """
+        if verifying_key is not None and not isinstance(verifying_key, str):
+                raise InvalidVerifyingKey("'verifying_key' must be a string")
+
+        if signing_key is not None:
+            if not isinstance(signing_key, str):
+                raise InvalidSigningKey("'signing_key must be a string")
+            if verifying_key is None:
+                raise InvalidVerifyingKey(("'verifying_key' must be given if"
+                                           "'signing_key' is given"))
```

Comment:
Using `.transfer` does not require a `verifying_key`, so why not allow a user to only supply a `signing_key`?